### PR TITLE
add support for Broadcom Lawful Intercept header

### PR DESCRIPTION
**Pull request by hannesgredler**

### CODE REVIEW

**Comment by guyharris** [comment]
Unlike Wireshark, we're still supporting both autotools and CMake, so you need to update Makefile.in by adding `print-bcm-li.c` to the list in `LIBNETDISSECT_SRC`.

**Comment by hannesgredler** [comment]
Thanks Guy - fixed.

**Comment by fxlb** [comment]
1) The function is defined "u_int bcm_li_print()" but the return value is not used.
Should it be defined "void bcm_li_print()" ?

2) Could you add a test capture file in tests/ directory ?

3) Trailing tabs in lines 74 and 95.

**Comment by hannesgredler** [comment]
Thanks - review comments pushed;

**Comment by fxlb** [comment]
Thanks, I suggest this patch to have output references, with the bcm-li.out and bcm-li-v.out built via
`./update-test.sh bcm-li` and `./update-test.sh bcm-li-v`
```patch
diff --git a/tests/TESTLIST b/tests/TESTLIST
index 150effb9..6f79ad6f 100644
--- a/tests/TESTLIST
+++ b/tests/TESTLIST
@@ -39,6 +39,10 @@ brcmtag		brcm-tag.pcap		brcm-tag.out
 brcmtag-e	brcm-tag.pcap		brcm-tag-e.out	-e
 brcmtagprepend	brcm-tag-prepend.pcap	brcm-tag-prepend.out -e
 
+# Broadcom LI
+bcm-li bcm-li.pcap bcm-li.out
+bcm-li-v bcm-li.pcap bcm-li-v.out -v
+
 # Marvell DSA tag tests
 dsa		dsa.pcap		dsa.out
 dsa-e		dsa.pcap		dsa-e.out	-e
```

**Comment by hannesgredler** [comment]
thanks - added;

File: print-bcm-li.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		ip6_print(ndo, bp, length);
+	    } else {
+		ND_PRINT("unknown payload");
+	    }
+	    break;
+
+	default:
+	    break;
+	}
+
```
Comment (fxlb) [inline print-bcm-li.c:123]: Missing return

Comment (fxlb) [reply print-bcm-li.c:123]: Adding this return implies updating `bcm-li-v.out`.

Comment (hannesgredler) [reply print-bcm-li.c:123]: done !

**Comment by fxlb** [comment]
@hannesgredler
Do you want to keep your PR like it is or to rebase it to have only one commit with `git rebase -i HEAD~7` using `fixup` or `squash` for the commits following the first one ?

**Comment by hannesgredler** [comment]
rebased and squashed.

**Comment by fxlb** [comment]
Rebased, merged, thank you !

**Pull request closed without merging**